cmd/vulcan-retirejs/main.go

```diff
@@ -363,38 +375,31 @@ func writeFile(fileName string, contents string) error {
 	return nil
 }
 
-func downloadFromUrl(url string) error {
-	filePath := getFilePath(url)
-	logger.Infof("Downloading %s to %s", url, filePath)
-	response, err := http.Get(url)
+func downloadFromUrl(URL string) error {
+	filename := ""
+	u, err := url.ParseRequestURI(URL)
+	if err == nil {
+		path := u.EscapedPath()
+		tokens := strings.Split(strings.Trim(path, "/"), "/")
+		filename = tokens[len(tokens)-1]
+	}
+	// NOTE: a random UUID is appended to avoid file path clashing with other
+	// previously downloaded scripts (e.g. '/a/jquery.min.js' and
+	// '/b/jquery.min.js').
+	filePath := fmt.Sprintf("%s/%s_%s.js", jsPath, filename, uuid.NewV4().String())
```

Comment:
It's not a big deal, but, If I'm getting this correctly, the `url` param can be an invalid URI, in which case, the function `url.ParseRequestURI` will return and error and cause this var `filePath` to get a value like: `temp/_UUID.js` which is a little bit odd. It is true that then call `response, err := client.Get(URL)` will return an error and that value will be discarded. But I'm curious why you didn't want to just do it in the more usual way:
```go
u, err := url.ParseRequestURI(URL)
	if err != nil {
		return fmt.Errorf("error invalid url %s: %v, URL, err)
	}
path := u.EscapedPath()
tokens := strings.Split(strings.Trim(path, "/"), "/")
filename = tokens[len(tokens)-1]
filePath := fmt.Sprintf("%s/%s_%s.js", jsPath, filename, uuid.NewV4().String())
``` 
Or if the objective is to ignore the file if the url is malformed do something like this:
```go
u, err := url.ParseRequestURI(URL)
	if err != nil {
            logger.Infof("skipping invalid url %s: %v" URL, err)
            return nil
	}
```

Reply:
Yes, you are totally right. When I started changing the code this part was in a function that wasn't returning errors, so I adapted the code to work for that situation. But now it doesn't make sense. Gonna apply your first suggestion.

Reply:
Fixed in [eca3732](https://github.com/adevinta/vulcan-checks/pull/381/commits/eca3732bd6e9ba06b6e1847bf2a1efc20762a99e)